tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -6,72 +6,76 @@ import (
-// svc should remain private since secretes should not be exposed
-type svc struct {
-	// service account for AWS EC2
+// cloudProviderInfo is an provider specific struct which contains clients for EC2, IAM,
+// and the existing OpenShift cluster that is running on EC2.
+// This is an implementation of the Cloud interface.
+type cloudProviderInfo struct {
+	// A client for EC2.
 	EC2 *ec2.EC2
-	// service account for AWS IAM
+	// A client for IAM.
 	IAM *iam.IAM
-	// client of the existing cluster
-	clientset *client.Clientset
-	// infraID is the readable unique infrastructure name generated by openshift-install for new infrastructures
-	infraID *string
-	// dir is the path to where created instance and security group information is stored
-	dir *string
+	// clientset is the client of the existing OpenShift cluster.
+	Clientset *openshift.Client
+	// resourceTrackerDir is the path to where created instance and security group information is stored.
+	resourceTrackerDir string
 }
 
-// New is the factory of AWS Cloud interface
-// It takes the generic type of information and returns an interface containing information specific to AWS
-func New(clientset *client.Clientset, infrastructure *v1.Infrastructure, credentialPath, credAccount, dir *string) (cloudprovider.Cloud, error) {
-	region := infrastructure.Status.PlatformStatus.AWS.Region
-	infraId := infrastructure.Status.InfrastructureName
-	session, err := NewSession(credentialPath, credAccount, &region)
+// New takes the generic type of information and returns an interface containing cloudProviderInfo specific to AWS.
+// credentialPath is a directory information to AWS credentials file.
+// credentialAccountID is the account name the user uses to create VM instance.
+// resourceTrackerDir is the path to where created instance and security group information is stored.
+func New(clientset *openshift.Client, credentialPath, credentialAccountID, resourceTrackerDir string) (cloudprovider.Cloud,
+	error) {
+	provider, err := clientset.GetCloudProvider()
 	if err != nil {
 		return nil, err
 	}
-	return &svc{ec2.New(session, aws.NewConfig()),
+	session, err := NewSession(credentialPath, credentialAccountID, provider.AWS.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &cloudProviderInfo{ec2.New(session, aws.NewConfig()),
 		iam.New(session, aws.NewConfig()),
 		clientset,
-		&infraId,
-		dir,
+		resourceTrackerDir,
 	}, nil
 }
 
-// NewSession uses AWS credentials to create a session for interacting with AWS EC2
-// It returns an AWS session
-func NewSession(credPath, credAccount, region *string) (*sessionaws.Session, error) {
-	if _, err := os.Stat(*credPath); os.IsNotExist(err) {
-		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", *credPath)
+// NewSession uses AWS credentials to create and returns a session for interacting with EC2.
+func NewSession(credentialPath, credentialAccountID, region string) (*awssession.Session, error) {
```

Comment:
Is there a reason to export this function? If not, please call it `newSession` as it seems to be a local helper.